packages/boilerplate/server/webpack.config.babel.js

```diff
@@ -40,17 +40,15 @@ export default (env) => {
       strictExportPresence: true, // If you import something that isn't exported
       rules: [
         {
-          test: /\.js$/,
-          exclude: /node_modules/,
+          test: /\.jsx?$/,
+          // Skip node_modules, with the exception of packages in the monorepo
+          exclude: /[\\/]node_modules[\\/](?!@respond-framework)/,
           use: {
             loader: 'babel-loader',
             options: {
               cacheDirectory: true,
-              plugins: [
-                'react-hot-loader/babel',
-                '@babel/syntax-dynamic-import',
-                'universal-import',
-              ],
+              rootMode: 'upward',
+              envName: 'webpack',
```

Comment:
you've done ur research, can you explain what upward and these env names are exactly doing. This seems very interesting and i rarely don't know a config. Id love for you to point me to documentation, or just hit me with a reply. 
I like learning new things and this is definitely more sophisticated than others I've seen.

Reply:
The `rootMode` is to solve the problem described here: https://babeljs.io/docs/en/config-files#project-wide-configuration

Basically it means "Babel, I know there is a package.json in this directory, but I want you to keep searching higher in the directory tree because there is a `babel.config.js` there". Its a bit of extra args when calling babel, but the benefit is not having to have those `.babelrc`/`babel.config.js` in each subpackage for no reason other than to re-export the root one.

Reply:
The `envName` is just a different babel config - basically in the root `babel.config.js` it uses different plugins based on the env: https://github.com/respond-framework/rudy/pull/6/files#diff-16eb920d570e18c01c5d4e561276dd6aR4

In this case, its so that `universal-import`, `syntax-dynamic-import` and co can be used just for the webpack compilation, without introducing potential problems in other code